server/blob/blob-objectstorage/src/main/java/org/apache/james/blob/objectstorage/ObjectStorageDumbBlobStore.java

```diff
@@ -0,0 +1,205 @@
+import org.apache.james.blob.api.ObjectNotFoundException;
+import org.apache.james.blob.api.ObjectStoreException;
+import org.apache.james.blob.api.ObjectStoreIOException;
+import org.apache.james.blob.objectstorage.aws.AwsS3AuthConfiguration;
+import org.apache.james.blob.objectstorage.aws.AwsS3ObjectStorage;
+import org.apache.james.blob.objectstorage.swift.SwiftKeystone2ObjectStorage;
+import org.apache.james.blob.objectstorage.swift.SwiftKeystone3ObjectStorage;
+import org.apache.james.blob.objectstorage.swift.SwiftTempAuthObjectStorage;
+import org.jclouds.blobstore.domain.Blob;
+import org.jclouds.blobstore.domain.StorageMetadata;
+import org.jclouds.blobstore.domain.StorageType;
+
+import com.github.fge.lambdas.Throwing;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import com.google.common.io.ByteSource;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
+public class ObjectStorageDumbBlobStore implements DumbBlobStore {
+    private static final int BUFFERED_SIZE = 256 * 1024;
+    public static final boolean LAZY = false;
+
+    private final BucketName defaultBucketName;
+    private final org.jclouds.blobstore.BlobStore blobStore;
+    private final BlobPutter blobPutter;
+    private final PayloadCodec payloadCodec;
+    private final ObjectStorageBucketNameResolver bucketNameResolver;
+
+    ObjectStorageDumbBlobStore(BucketName defaultBucketName,
+                               org.jclouds.blobstore.BlobStore blobStore,
+                               BlobPutter blobPutter,
+                               PayloadCodec payloadCodec, ObjectStorageBucketNameResolver bucketNameResolver) {
+        this.defaultBucketName = defaultBucketName;
+        this.blobStore = blobStore;
+        this.blobPutter = blobPutter;
+        this.payloadCodec = payloadCodec;
+        this.bucketNameResolver = bucketNameResolver;
+    }
+
+    public static ObjectStorageDumbBlobStoreBuilder builder(SwiftTempAuthObjectStorage.Configuration testConfig) {
+        return SwiftTempAuthObjectStorage.blobStoreBuilder(testConfig);
+    }
+
+    public static ObjectStorageDumbBlobStoreBuilder builder(SwiftKeystone2ObjectStorage.Configuration testConfig) {
+        return SwiftKeystone2ObjectStorage.blobStoreBuilder(testConfig);
+    }
+
+    public static ObjectStorageDumbBlobStoreBuilder builder(SwiftKeystone3ObjectStorage.Configuration testConfig) {
+        return SwiftKeystone3ObjectStorage.blobStoreBuilder(testConfig);
+    }
+
+    public static ObjectStorageDumbBlobStoreBuilder builder(AwsS3AuthConfiguration testConfig) {
+        return AwsS3ObjectStorage.blobStoreBuilder(testConfig);
+    }
+
+    @PreDestroy
+    public void close() throws IOException {
+        blobStore.getContext().close();
+        blobPutter.close();
+    }
+
+    private boolean isItABigStream(InputStream bufferedData) throws IOException {
```

Comment:
No longer needed. 

Remove the save(InputSteam) methodfrom the dumb blob store api, and get the final size from the byteSource.

Also, the caller passes the blovId, this code can be simplified compared to the previous version of it.